platforms/core-configuration/configuration-cache/src/integTest/groovy/org/gradle/internal/cc/impl/isolated/IsolatedProjectsToolingApiModelQueryIntegrationTest.groovy

```diff
@@ -139,6 +139,47 @@ class IsolatedProjectsToolingApiModelQueryIntegrationTest extends AbstractIsolat
         outputContains("Execution of dummyTask")
     }
 
+    def "can cache models with tasks using internal option"() {
+        given:
+        withSomeToolingModelBuilderPluginInBuildSrc()
+        settingsFile << """
+            include("a")
+            include("b")
+        """
+        buildFile << """
+            plugins.apply(my.MyPlugin)
+
+            tasks.register("dummyTask") {
+                println("Configuration of dummyTask")
+                doLast {
+                    println("Execution of dummyTask")
+                }
+            }
+        """
+
+        when:
+        withIsolatedProjects("-Dorg.gradle.internal.isolated-projects.configure-on-demand.tasks=true")
+        fetchModel(SomeToolingModel, ":dummyTask")
+
+        then:
+        fixture.assertModelStored {
+            projectsConfigured(":buildSrc", ":")
+            modelsCreated(":")
```

Comment:
❔ Does it store the work graph too?

Reply:
It does store the work graph. On a later cache hit the work graph is also loaded for task execution.

Your question made me realize our fixtures are not checking strict enough invariants. I'll fix them in the same PR.